### PR TITLE
test: test replicas

### LINKED ISSUE #7: Verify the usage of replicas
As a user of ESIndexer, I would like to have the indexer return same results with the `replicas>1`.

```python
from jina import Flow
from executor import ElasticSearchIndexer
from docarray import Document
import numpy as np

n_dim = 10

f = Flow().add(
    uses=ElasticSearchIndexer,
    uses_with={'index_name': 'test1', 'n_dim': n_dim},
)

docs_index = [Document(embedding=np.random.random(n_dim)) for _ in range(100)]

docs_query = docs_index[:10]

with f:
    f.post(
        on='/index',
        inputs=docs_index,
    )

    docs_without_replicas = f.post(
        on='/search',
        inputs=docs_query,
    )

f_with_replicas = Flow().add(
    uses=ElasticSearchIndexer,
    uses_with={'index_name': 'test1', 'n_dim': n_dim},
    replicas=4
)

with f_with_replicas :
    f_with_replicas .post(
        on='/index',
        inputs=docs_index,
    )

    docs_with_replicas = f_with_replicas.post(
        on='/search',
        inputs=docs_query,
    )

assert docs_without_replicas == docs_with_replicas
```

### PR DESCRIPTION
**Pull request by winstonww**
This PR test search results of elastic search indexer with and without replicas.

resolved #7

### CODE REVIEW

**Review by JoanFM — COMMENTED** [review]
For even more security, can u put `request_size=1` in both `index` and `search` calls?

File: tests/integration/test_replicas.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    )
+    
+    docs_index = [Document(embedding=np.random.random(n_dim)) for _ in range(1000)]
+    
+    docs_query = docs_index[:100]
+    
+    with f:
+        f.post(
+            on='/index',
+            inputs=docs_index,
```
Comment (JoanFM) [inline tests/integration/test_replicas.py]:
```suggestion
            inputs=docs_index,
            request_size=1
```

File: tests/integration/test_replicas.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+        )
+    
+        docs_without_replicas = f.post(
+            on='/search',
+            inputs=docs_query,
```
Comment (JoanFM) [inline tests/integration/test_replicas.py]:
```suggestion
            inputs=docs_query,
            request_size=1
```

File: tests/integration/test_replicas.py [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    f_with_replicas = Flow().add(
+        uses=ElasticSearchIndexer,
+        uses_with={'index_name': 'test1', 'n_dim': n_dim},
+        replicas=4
+    )
+    
+    with f_with_replicas :
+        docs_with_replicas = f_with_replicas.post(
+            on='/search',
+            inputs=docs_query,
```
Comment (JoanFM) [inline tests/integration/test_replicas.py]:
```suggestion
            inputs=docs_query,
            request_size=1
```

**Review by JoanFM — CHANGES_REQUESTED** [review]
Besides suggestions, please also have a test where u also `reindex` with `replicas`

File: tests/integration/test_replicas.py (hunk 1 above)
Comment (winstonww) [reply tests/integration/test_replicas.py]: Addressed.

File: tests/integration/test_replicas.py (hunk 2 above)
Comment (winstonww) [reply tests/integration/test_replicas.py]: Addressed.

File: tests/integration/test_replicas.py (hunk 3 above)
Comment (winstonww) [reply tests/integration/test_replicas.py]: Thanks, a new test to reindex with replicas is added.

**Review by JoanFM — APPROVED** [review]

**Pull request merged**